### ISSUE TITLE
json.decode() into primitive types

### ISSUE THREAD

**Issue by czkz** [vlang/v#11492]
**V doctor:**
```
OS: linux, "Void Linux"
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5 CPU         650  @ 3.20GHz
CC version: cc (GCC) 10.2.1 20201203

getwd: /home/dek/test
vmodules: /home/dek/.vmodules
vroot: /home/dek/.v
vexe: /home/dek/.v/v
vexe mtime: 2021-09-14 13:56:11
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 0370d4f.0c0e28d

Git version: git version 2.33.0
Git vroot status: weekly.2021.36-61-g0c0e28df
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg bug.v`
```v
import json

fn good() {
	json.decode([]int, '1')
}

fn good_compilation_error() {
	// json.decode(rune, '1')
	// json.decode(isize, '1')
	// json.decode(usize, '1')
	// json.decode(voidptr, '1')
	// json.decode(any, '1')
}

// Any single line breaks v
fn bad() {
	// json.decode(bool, '1')
	// json.decode(string, '1')

	// json.decode(i8, '1')
	// json.decode(i16, '1')
	json.decode(int, '1')
	// json.decode(i64, '1')

	// json.decode(byte, '1')
	// json.decode(u16, '1')
	// json.decode(u32, '1')
	// json.decode(u64, '1')

	// json.decode(f32, '1')
	// json.decode(f64, '1')

	// json.decode(int, '{"a":"1"}')
}


fn main() {
	println('Hello, World!')
}

```

**What did you expect to see?**

Either a successful compilation or a compilation error.
Note: https://json.org says that numbers (`1`) and bools (`true`) are valid json.

**What did you see instead?**
```
==================
/usr/lib64/crt1.o: error: Invalid relocation entry [14] '.rela.debug_info' @ 000004f6
tcc: error: file 'crt1.o' not found
/tmp/v/bug.14262768652375047328.tmp.c:10722: error: 'Option__v_int' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

**Comment by enghitalo** [comment]
```v
module main
import x.json2 as json

fn main() {
	// println(json.decode[[]int]('1') or {0}.str()) //error: cannot convert 'int' to 'struct array'

	println(json.decode[rune]('1')!)
	println(json.decode[isize]('1')!)
	println(json.decode[usize]('1')!)
	// println(json.decode[voidptr]('1')!)//error: mismatched types `voidptr` and `voidptr`
	// println(json.decode[any]('1')!) //cannot use `any` type here, `any` will be implemented in V 0.4

	println(json.decode[bool]('1')!)
	println(json.decode[string]('1')!)
	println(json.decode[i8]('1')!)
	println(json.decode[i16]('1')!)
	println(json.decode[i64]('1')!)
	println(json.decode[byte]('1')!)
	println(json.decode[u16]('1')!)
	println(json.decode[u16]('1')!)
	println(json.decode[u32]('1')!)
	println(json.decode[u64]('1')!)
	println(json.decode[f32]('1')!)
	println(json.decode[f64]('1')!)
	println(json.decode[int]('1')!)

}
```

```bash

0
0
false
println(NIL)
0
0
0
0
0
0
0
0
0.0
0.0
0
```

**Comment by spytheman** [comment]
The code in the issue no longer produces cgen errors, but checker ones. There is also a known workaround by using a substitution with `x.json2`.
